var/spack/repos/builtin/packages/spiral/package.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Spiral(CMakePackage):
+    """SPIRAL is a program generation system for linear transforms and other
+    mathematical functions that produces very high performance code for a wide
+    spectrum of hardware platforms."""
+
+    homepage = "https://spiral.net"
+    url      = "https://github.com/spiral-software/spiral-software/archive/8.1.2.tar.gz"
+
+    maintainers = ['spiralgen']
+
+    version('8.1.2', sha256='506f1dbf923aa1c9f19f05444fa947085715eef37c9d2494d133fcaaa1dd50bc')
+
+    extendable = True
+
+    # No dependencies.
+
+    def build(self, spec, prefix):
+        with working_dir(self.build_directory):
+            make('all')
+            make('install/local')
+
+        # For some reason the make install/local doesn't seem to install
+        # the gap exe...though it does work if run manually
+        gapfil = join_path(self.build_directory, 'gap/src/gap')
+        dest = join_path(self.stage.source_path, 'gap/bin')
+        install(gapfil, dest)
+
+        @on_package_attributes(run_tests=True)
+        def test(self):
+            make('test')
```

Comment:
This can't be inside of a function, it needs to be its own function. The default for `CMakePackage` base class is to run `make test` or `make check` after build, so this function may not even be needed.

Reply:
This function is not needed; the default for CMakePackage works fine, removed.